File: src/src_core/fmg/core/mosaic/AMosaicImageController.java
package fmg.core.mosaic;

import java.util.concurrent.ThreadLocalRandom;

import fmg.core.img.*;
import fmg.data.view.draw.PenBorder;

/** MVC: mosaic image controller. Base implementation */
public abstract class AMosaicImageController<TImage,        TMosaicView extends AMosaicView<TImage, Void, MosaicAnimatedModel<Void>>>
                   extends AMosaicController<TImage, Void,  TMosaicView,                                  MosaicAnimatedModel<Void>>
              implements IAnimatedController<TImage,        TMosaicView,                                  MosaicAnimatedModel<Void>>
{
   private final AnimatedInnerController<TImage, TMosaicView, MosaicAnimatedModel<Void>> _innerController;

   public AMosaicImageController(TMosaicView view) {
      super(view);
      MosaicAnimatedModel<Void> model = getModel();
      _innerController = new AnimatedInnerController<>(model);
      addModelTransformer(new MosaicRotateTransformer());

      PenBorder pen = model.getPenBorder();
      pen.setColorLight(pen.getColorShadow());
      model.getBackgroundFill().setMode(1 + ThreadLocalRandom.current().nextInt(model.getCellAttr().getMaxBackgroundFillModeValue()));
   }

   @Override
   public void addModelTransformer(IModelTransformer transformer) {
      _innerController.addModelTransformer(transformer);
   }
   @Override
   public void removeModelTransformer(Class<? extends IModelTransformer> transformerClass) {
      _innerController.removeModelTransformer(transformerClass);
   }

}

File: src/src_core/fmg/core/mosaic/AMosaicImageController.cs
using System;
using fmg.common;
using fmg.core.img;

namespace fmg.core.mosaic {

   /// <summary> MVC: mosaic image controller. Base implementation </summary>
   public abstract class AMosaicImageController<TImage,          TMosaicView>
                            : AMosaicController<TImage, Nothing, TMosaicView, MosaicAnimatedModel<Nothing>>,
                            IAnimatedController<TImage,          TMosaicView, MosaicAnimatedModel<Nothing>>
      where TImage : class
      where TMosaicView : AMosaicView<TImage, Nothing, MosaicAnimatedModel<Nothing>>
   {

      private readonly AnimatedInnerController<TImage, TMosaicView, MosaicAnimatedModel<Nothing>> _innerController;

      public AMosaicImageController(TMosaicView view)
         : base(view)
      {
         var model = Model;
         _innerController = new AnimatedInnerController<TImage, TMosaicView, MosaicAnimatedModel<Nothing>>(model);
         AddModelTransformer(new MosaicRotateTransformer<TImage>());

         var pen = model.PenBorder;
         pen.ColorLight = pen.ColorShadow;
         model.BkFill.Mode = 1 + ThreadLocalRandom.Current.Next(model.CellAttr.GetMaxBackgroundFillModeValue());
      }

      public void AddModelTransformer(IModelTransformer transformer) {
         _innerController.AddModelTransformer(transformer);
      }
      public void RemoveModelTransformer(Type /* extends IModelTransformer */ transformerClass) {
         _innerController.RemoveModelTransformer(transformerClass);
      }

   }

}
